File: backend/app/schemas/matchmaker.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MatchRecommendation(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    user_id: UUID
    full_name: str
    username: str | None = None
    college: str
    course: str
    year: int

    bio: str | None = None

    skills: list[str] = []
    preferred_roles: list[str] = []

    match_score: float

    matched_skills: list[str] = []
    matched_roles: list[str] = []


class MatchmakerResponse(BaseModel):
    recommendations: list[MatchRecommendation]

File: frontend/services/matchmaker.ts
import { api } from "@/lib/api";

export interface MatchRecommendation {
  user_id: string;
  full_name: string;
  username?: string | null;

  college?: string | null;
  course?: string | null;
  year?: number | null;

  bio?: string | null;

  skills: string[];
  preferred_roles: string[];

  match_score: number;

  matched_skills: string[];
  matched_roles: string[];
}

export interface MatchmakerResponse {
  recommendations: MatchRecommendation[];
}

export async function getMatchmakerRecommendations(
  limit: number = 10
): Promise<MatchRecommendation[]> {
  const response = await api.get<MatchmakerResponse>(
    "/matchmaker/recommendations",
    {
      params: {
        limit,
      },
    }
  );

  return response.data.recommendations;
}